worlds/tloz_oos/__init__.py

```diff
@@ -634,6 +645,23 @@ def build_item_pool_dict(self):
         item_pool_dict["Gasha Seed"] = required_gasha_seeds
         filler_item_count -= required_gasha_seeds
 
+        # Add the required rings
+        ring_copy = sorted(self.options.required_rings.value.copy())
+        for _ in range(len(ring_copy)):
+            ring_name = ring_copy.pop()
+            if ITEMS_DATA[ring_name]["classification"] == ItemClassification.filler:
+                ring_name = f"{ring_name}!USEFUL"
+            item_pool_dict[ring_name] = item_pool_dict.get(ring_name, 0) + 1
+            
+            if item_pool_dict["Random Ring"] > 0:
+                # Take from set ring pool first
+                item_pool_dict["Random Ring"] -= 1
+                if item_pool_dict["Random Ring"] <= 0:
+                    del item_pool_dict["Random Ring"]
```

Comment:
You don't need to do that, random rings entry being deleted will be handled later